src/gui/sdlmain.cpp

```diff
@@ -675,38 +712,59 @@ Bitu GFX_SetSize(Bitu width,Bitu height,Bitu flags,double scalex,double scaley,G
 			free(sdl.opengl.framebuf);
 		}
 		sdl.opengl.framebuf=0;
-		if (!(flags&GFX_CAN_32)) goto dosurface;
-		int texsize=2 << int_log2(width > height ? width : height);
+		if (!(flags & GFX_CAN_32))
+			goto dosurface;
+		int texsize = 2 << int_log2(width > height ? width : height);
 		if (texsize>sdl.opengl.max_texsize) {
 			LOG_MSG("SDL:OPENGL: No support for texturesize of %d, falling back to surface",texsize);
 			goto dosurface;
 		}
-		SDL_GL_SetAttribute( SDL_GL_DOUBLEBUFFER, 1 );
-#if SDL_VERSION_ATLEAST(1, 2, 11)
-		SDL_GL_SetAttribute( SDL_GL_SWAP_CONTROL, 0 );
-#endif
-		GFX_SetupSurfaceScaled(SDL_OPENGL,0);
-		if (!sdl.surface || sdl.surface->format->BitsPerPixel<15) {
-			LOG_MSG("SDL:OPENGL: Can't open drawing surface, are you running in 16bpp (or higher) mode?");
+		SDL_GL_SetAttribute(SDL_GL_DOUBLEBUFFER, 1);
+		GFX_SetupWindowScaled(sdl.desktop.want_type);
+		/* We may simply use SDL_BYTESPERPIXEL
+		here rather than SDL_BITSPERPIXEL   */
+		if (!sdl.window || SDL_BYTESPERPIXEL(SDL_GetWindowPixelFormat(sdl.window))<2) {
+			LOG_MSG("SDL:OPENGL:Can't open drawing window, are you running in 16bpp(or higher) mode?");
 			goto dosurface;
 		}
+		sdl.opengl.context = SDL_GL_CreateContext(sdl.window);
+		if (sdl.opengl.context == NULL) {
+			LOG_MSG("SDL:OPENGL:Can't create OpenGL context, falling back to surface");
+			goto dosurface;
+		}
+		/* Sync to VBlank if desired */
+		SDL_GL_SetSwapInterval(sdl.desktop.vsync ? 1 : 0);
```

Comment:
Allow vsync values to be `(-1, 0, 1)`, where `-1` is adaptive sync: https://wiki.libsdl.org/SDL_GL_SetSwapInterval

Reply:
I started implementation, but ATM I think it might be better to move it to a separate PR… This invocation here is used only for `output = opengl`, but it should be also tried for `texture`. Also, I see some inconsistencies between the documentation of this SDL function and behaviour I see during implementation… (but maybe that's just OpenGL).

Reply:
Yes; also after more testing I agree it makes sense to move follow up work to the next SDL2 PR. 
Adaptive vsync (`-1`) is accepted and "works", but after more testing, vsync in hardware-accelerated modes generates significant video lag (atleast on my side.. will follow up w/ an issue).